Sources/PackageLoading/ModuleMapGenerator.swift

```diff
@@ -152,9 +153,9 @@ public struct ModuleMapGenerator {
         stream <<< "module \(target.c99name) {\n"
         switch type {
         case .header(let header):
-            stream <<< "    umbrella header \"\(header.pathString)\"\n"
+            stream <<< "    umbrella header \"\(header.pathString.replacingOccurrences(of: "\\", with: "\\\\"))\"\n"
         case .directory(let path):
-            stream <<< "    umbrella \"\(path.pathString)\"\n"
+            stream <<< "    umbrella \"\(path.pathString.replacingOccurrences(of: "\\", with: "\\\\"))\"\n"
```

Comment:
should we do this when constructing the path instead of patching it here?

Reply:
We cannot do that when constructing the path I think.  As I state in the commit message, `.pathString` is used both for the file system representation which is required for the construction of the command and for the string representation which needs to be escaped.  That distinction is part of `URL` but is not part of the representation in tools-support-core.

Reply:
I see, maybe we should add an "xxxEscaped" accessor then? Just so we dont have to repeat this in other spots

Reply:
Sure, I don't think that's terrible, however, I've not had to add that to any other site as of yet - and I'm at the point where I can get through a build of s-p-m.

Reply:
The extra escape is part of encoding the path into the module map file — it is not an inherent part of the path.  So I agree with @compnerd here — the paths themselves should never contain the escaping.  However, I would have thought for sure that we had common code to escape strings for various intents (for shell scripts, etc) and I will see if there is anything more common that we can use here.  Note that "xxxEscaped" isn't enough, because the escaping can be different for different intents (shells, URL parsers, and various other decoders all treat different characters as special, so the escaping needs to take the context into account).  In this case, I'm guessing that the paths in the module map are escaped the same way as paths in preprocessor directives, but we'd need to look at the compiler to know for sure.

Reply:
right. in "xxxEscaped" I meant things like "shellEscaped", "modulemapEscaped" etc

Reply:
Ah, got it.  Yup, that makes sense.

Reply:
Even if this is the first and so far only use of this escaping here, I think it would make sense to add it as part of this PR.  Otherwise we just end up with a lot of special cases to clean up later.

Reply:
@compnerd lets get this factored out so we can merge

Reply:
@abertelrud - as an extension in s-p-m?  Cause there are a bunch of t-s-c patches that need to get merged first before I can really start messing with that.

Reply:
I was thinking of `StringConversions.swift`, but you're right, that's in ToolSupportCore these days.  I would say just a fileprivate extension in the `ModuleMapGenerator.swift` would be relatively simple and would factor it out somewhat.  That could then later be moved to someplace general-purpose.  Escaping for modulemaps is also fairly specialized, so even in the long run it might make sense to keep close to the module map code.  But having at least a file private extension would be good, I think.